release.py

```diff
@@ -0,0 +1,694 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Release automation script."""
+
+from __future__ import annotations
+
+import argparse
+import logging
+import os
+import re
+import subprocess
+from datetime import datetime
+from pathlib import Path
+
+import github
+import github.GitRelease
+import github.Repository
+
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger('release')
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+if 'GITHUB_TOKEN' not in os.environ:
+    raise SystemExit('Environment variable GITHUB_TOKEN not set.')
+
+auth = github.Auth.Token(os.environ['GITHUB_TOKEN'])
+gh_client = github.Github(auth=auth)
+
+
+VERSION_REGEX = r'(\d+\.\d+\.\d+(?:\.dev\d+)?)'
+VERSION_FILES = {
+    'ops/src': Path('ops/version.py'),
+    'ops/pyproject': Path('pyproject.toml'),
+    'testing': Path('testing/pyproject.toml'),
+    'tracing': Path('tracing/pyproject.toml'),
+    'uvlock': Path('uv.lock'),
+}
+
+
+def get_latest_release_tag(repo: github.Repository.Repository, branch_name: str) -> str | None:
+    """Get the latest release tag from the repository."""
+    releases = repo.get_releases()
+
+    maintenance_branch_version = (
+        branch_name.removesuffix('-maintenance') if branch_name.endswith('-maintenance') else None
+    )
+
+    for release in releases:
+        if release.draft:
+            continue
+
+        if maintenance_branch_version and maintenance_branch_version not in release.tag_name:
+            continue
+
+        return release.tag_name
+
+    return None
+
+
+def bump_minor_version(version: str) -> str:
+    """Bump minor version."""
+    major, minor, patch = map(int, version.split('.'))
+    return f'{major}.{minor + 1}.{patch}'
+
+
+def get_new_tag_for_release(
+    owner: str, repo: github.Repository.Repository, branch_name: str
+) -> str:
+    """Get a new tag for release.
+
+    Default value is generated by increasing the minor version of the latest release's tag.
+
+    Can be overridden with a user-provided version.
+    """
+    latest_tag = get_latest_release_tag(repo, branch_name)
+
+    suggested_tag = ''
+    if not latest_tag:
+        logger.info(f'No version tag found in branch "{branch_name}"')
+    else:
+        logger.info(f'Latest tag in branch "{branch_name}": {latest_tag}')
+        if not re.match(r'^\d+\.\d+\.\d+$', latest_tag):
+            logger.info('Latest tag is not in format X.Y.Z.')
+        else:
+            suggested_tag = bump_minor_version(latest_tag)
+            logger.info(f'Suggested new version: {suggested_tag}')
+
+    tag_prompt = f' (press enter to use the tag {suggested_tag})' if suggested_tag else ''
+    prompt = f"Input the new tag for the release{tag_prompt}, or 'c' to cancel:\n"
```

Comment:
I don't think we need special handling of "c" to cancel. It's common practice to press Ctrl-C to cancel these kind of scripts (which Python would handle automatically).